src/dotnet-grpc/Commands/AddFileCommand.cs

```diff
@@ -0,0 +1,79 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using Grpc.Dotnet.Cli.Internal;
+using Grpc.Dotnet.Cli.Options;
+
+namespace Grpc.Dotnet.Cli.Commands
+{
+    internal class AddFileCommand : CommandBase
+    {
+        public static Command Create()
+        {
+            var command = new Command(
+                name: "file",
+                description: "Add protobuf file reference(s) to the gRPC project.",
+                argument: new Argument<string[]>
+                {
+                    Name = "files",
+                    Description = "The protobuf file reference(s). These can be a path to glob for local protobuf file(s).",
+                });
+
+            command.AddOption(CommonOptions.ProjectOption());
+            command.AddOption(CommonOptions.ServiceOption());
+            command.AddOption(CommonOptions.AdditionalImportDirsOption());
+            command.AddOption(CommonOptions.AccessOption());
+
+            command.Handler = CommandHandler.Create<IConsole, FileInfo, Services, Access, string, string[]>(new AddFileCommand().AddFile);
+
+            return command;
+        }
+
+        public int AddFile(IConsole console, FileInfo? project, Services services, Access access, string additionalImportDirs, string[] files)
```

Comment:
It feels odd that console is passed in on a method and set on a property. I know you did it to avoid having to pass it around everywhere, and this method should only ever be called once.